src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -177,6 +177,70 @@ def _construct_search_ddf(
         ddf_partition_map = {pixel: i for i, pixel in enumerate(filtered_pixels)}
         return ddf_partition_map, search_ddf
 
+    def map_partitions(
+        self,
+        func: Callable[..., pd.DataFrame],
+        *args,
+        meta: pd.DataFrame | pd.Series | Dict | Iterable | Tuple | None = None,
+        include_pixel: bool = False,
+        **kwargs,
+    ) -> Self:
+        """Applies a function to each partition in the catalog.
+
+        The ra and dec of each row is assumed to remain unchanged.
+
+        Args:
+            func (Callable): The function applied to each partition, which will be called with:
+                `func(partition: pd.DataFrame, *args, **kwargs)` with the additional args and kwargs passed to
+                the `map_partitions` function. If the `include_pixel` parameter is set, the function will be
+                called with the `healpix_pixel` as the second positional argument set to the healpix pixel
+                of the partition as
+                `func(partition: pd.DataFrame, healpix_pixel: HealpixPixel, *args, **kwargs)`
+            *args: Additional positional arguments to call `func` with.
+            meta (pd.DataFrame | pd.Series | Dict | Iterable | Tuple | None): An empty pandas DataFrame that
+                has columns matching the output of the function applied to a partition. Other types are
+                accepted to describe the output dataframe format, for full details see the dask documentation
+                https://blog.dask.org/2022/08/09/understanding-meta-keyword-argument
+                If meta is None (default), LSDB will try to work out the output schema of the function by
+                calling the function with an empty DataFrame. If the function does not work with an empty
+                DataFrame, this will raise an error and meta must be set. Note that some operations in LSDB
+                will generate empty partitions, though these can be removed by calling the
+                `Catalog.prune_empty_partitions` method.
+            include_pixel (bool): Whether to pass the Healpix Pixel of the partition as a `HealpixPixel`
+                object to the second positional argument of the function
+            **kwargs: Additional keyword args to pass to the function. These are passed to the Dask DataFrame
+                `dask.dataframe.map_partitions` function, so any of the dask function's keyword args such as
+                `transform_divisions` will be passed through and work as described in the dask documentation
+                https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.map_partitions.html
+
+        Returns (Catalog):
```

Comment:
```suggestion
        Returns:
```

Autodoc rendering doesn't like this:

https://lsdb--290.org.readthedocs.build/en/290/autoapi/lsdb/catalog/dataset/healpix_dataset/index.html#lsdb.catalog.dataset.healpix_dataset.HealpixDataset.map_partitions